experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -0,0 +1,540 @@
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:energy}) = CL.total_energy(sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:surface_direct_albedo}) =
+    CL.surface_albedo(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:surface_diffuse_albedo}) =
+    CL.surface_albedo(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:water}) = CL.total_water(sim.integrator.u, sim.integrator.p)
+function Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:height_disp})
+    d_canopy = CL.displacement_height(sim.model.canopy, sim.integrator.u, sim.integrator.p)
+    d_snow = CL.displacement_height(sim.model.snow, sim.integrator.u, sim.integrator.p)
+    d_soil = CL.displacement_height(sim.model.soil, sim.integrator.u, sim.integrator.p)
+    return NamedTuple{(:canopy, :snow, :soil)}((d_canopy, d_snow, d_soil))
+end
+function Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:roughness_buoyancy})
+    z_0b_canopy = sim.model.canopy.parameters.z_0b
+    z_0b_snow = sim.model.snow.parameters.z_0b
+    z_0b_soil = sim.model.soil.parameters.z_0b
+    return NamedTuple{(:canopy, :snow, :soil)}((z_0b_canopy, z_0b_snow, z_0b_soil))
+end
+function Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:roughness_momentum})
+    z_0m_canopy = sim.model.canopy.parameters.z_0m
+    z_0m_snow = sim.model.snow.parameters.z_0m
+    z_0m_soil = sim.model.soil.parameters.z_0m
+    return NamedTuple{(:canopy, :snow, :soil)}((z_0m_canopy, z_0m_snow, z_0m_soil))
+end
+function Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:surface_temperature})
+    T_eff = sim.integrator.p.T_sfc
+    T_canopy = CL.surface_temperature(sim.model.canopy, sim.integrator.u, sim.integrator.p, sim.integrator.t)
+    T_snow = CL.surface_temperature(sim.model.snow, sim.integrator.u, sim.integrator.p, sim.integrator.t)
+    T_soil = CL.surface_temperature(sim.model.soil, sim.integrator.u, sim.integrator.p, sim.integrator.t)
+    return NamedTuple{(:effective, :canopy, :snow, :soil)}((T_eff, T_canopy, T_snow, T_soil))
+end
+
+# Update fields stored in land model sub-components
+function Interfacer.update_field!(sim::ClimaLandSimulation, ::Val{:area_fraction}, field)
+    parent(sim.area_fraction) .= parent(field)
+end
+function Interfacer.update_field!(sim::ClimaLandSimulation, ::Val{:diffuse_fraction}, field)
+    p = sim.integrator.p
+    parent(p.canopy.radiative_transfer.diffuse_fraction) .= parent(field)
+end
+
+"""
+    update_field!(sim::ClimaLandSimulation, ::Val{:turbulent_energy_flux}, fields::NamedTuple)
+
+Takes in a NamedTuple having fields `canopy`, `snow`, and `soil`, which in turn
+each contain fields `lhf` and `shf`, and updates each component of the land
+simulation with the corresponding LHF and SHF fields.
+"""
+function Interfacer.update_field!(sim::ClimaLandSimulation, ::Val{:turbulent_energy_flux}, fields::NamedTuple)
```

Comment:
When is this function called?

Reply:
This is used by the combined state fluxes option, called here https://github.com/CliMA/ClimaCoupler.jl/blob/710a9f5c703ae40d7caea88b10cbc9339a630b3f/src/FieldExchanger.jl#L176-L181

When using partitioned state fluxes (which is what we're focusing on for the full land model), the component model's fluxes get updated directly

Reply:
I guess since we aren't supporting combined surface fluxes for the full land model, we can remove these two functions